### PR TITLE
Rapid7.Metasploit version 6.1.44.1

### PR DESCRIPTION
**Pull request by AlexDev404**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.1 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.1.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/61780)

### CODE REVIEW

**Comment by AlexDev404** [comment]
## NOTE TO REVIEWER
When installing, AV detects as virus. Not sure if it may cause any issues with adding it to the repository.

**Comment by AlexDev404** [comment]
Would linking another binary work if it's not detected as malware?

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
> Would linking another binary work if it's not detected as malware?

WinGet package repository only accepts applications that passes automated tests for static analysis and dynamic analysis. So if this is a false positive then you'll need to contact anti-virus vendors to mark this application as "not virus / malware" so the anti-malware definitions can be updated with the false positive removed.

You can submit another application that isn't considered as malware / PUA / virus and it'll pass as long as it has support for silent installation, aren't any install-time dependencies, and aren't any install-time drivers blocking the installer from running silently (unattended mode).

Also: image

File: manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+PackageIdentifier: Rapid7.Metasploit
+PackageVersion: 6.1.44.1
+PackageLocale: en-US
+Publisher: Rapid7, Inc.
+PublisherUrl: https://www.rapid7.com/
+PublisherSupportUrl: https://www.rapid7.com/for-customers/
+PrivacyUrl: https://www.rapid7.com/privacy-policy/
+Author: Rapid7, Inc.
+PackageName: Metasploit Framework
```
Comment (ItzLevvie) [inline manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.locale.en-US.yaml:12]:
```suggestion
PackageName: Metasploit-framework
```
This must match Control Panel.

File: manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Created using wingetcreate 1.0.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.1.0.schema.json
+
+PackageIdentifier: Rapid7.Metasploit
+PackageVersion: 6.1.44.1
+PackageLocale: en-US
+Publisher: Rapid7, Inc.
```
Comment (ItzLevvie) [inline manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.locale.en-US.yaml:7]:
```suggestion
Publisher: Rapid7 Release Engineering <r7_re@rapid7.com>
```
This must match Control Panel.

File: manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+PackageIdentifier: Rapid7.Metasploit
+PackageVersion: 6.1.44.1
+Installers:
+- InstallerLocale: en-US
+  Architecture: x64
+  InstallerType: wix
+  InstallerUrl: https://windows.metasploit.com/metasploitframework-latest.msi
+  InstallerSha256: 4362A029C314586FB2CBF134F8AA6DA40DDC1324701920A0535BE1FED6340482
+  ProductCode: '{19D66627-55BB-41D4-A9F6-F94A408E66F2}'
```
Comment (ItzLevvie) [inline manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.installer.yaml:12]:
```suggestion
  ProductCode: '{63128707-33F1-4DEA-B4E0-631E9F25BBE8}'
```

File: manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: Rapid7.Metasploit
+PackageVersion: 6.1.44.1
+Installers:
+- InstallerLocale: en-US
+  Architecture: x64
+  InstallerType: wix
+  InstallerUrl: https://windows.metasploit.com/metasploitframework-latest.msi
+  InstallerSha256: 4362A029C314586FB2CBF134F8AA6DA40DDC1324701920A0535BE1FED6340482
```
Comment (ItzLevvie) [inline manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.installer.yaml:11]:
```suggestion
  InstallerSha256: D243B3B7266039EEA4294FA81EEA0B465650AE5CBAF866DB9D124EF2A0AE4AB7
```

File: manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+PackageVersion: 6.1.44.1
+PackageLocale: en-US
+Publisher: Rapid7, Inc.
+PublisherUrl: https://www.rapid7.com/
+PublisherSupportUrl: https://www.rapid7.com/for-customers/
+PrivacyUrl: https://www.rapid7.com/privacy-policy/
+Author: Rapid7, Inc.
+PackageName: Metasploit Framework
+PackageUrl: https://www.metasploit.com/
+License: BSD 3-clause
```
Comment (ItzLevvie) [inline manifests/r/Rapid7/Metasploit/6.1.44.1/Rapid7.Metasploit.locale.en-US.yaml:14]:
```suggestion
License: BSD-3-Clause
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Review by ItzLevvie — DISMISSED** [review]

**Comment by AlexDev404** [comment]
> > Would linking another binary work if it's not detected as malware?
> 
> WinGet package repository only accepts applications that passes automated tests for static analysis and dynamic analysis. So if this is a false positive then you'll need to contact anti-virus vendors to mark this application as "not virus / malware" so the anti-malware definitions can be updated with the false positive removed.
> 
> You can submit another application that isn't considered as malware / PUA / virus and it'll pass as long as it has support for silent installation, aren't any install-time dependencies, and aren't any install-time drivers blocking the installer from running silently (unattended mode).
> 
> Also: image

Hi Levvie,
You stated that in order to remove this false-positive I would need to contact the anti-malware vendors to remove this false positive. Does this mean I have to contact each one of them individually?

**Comment by AlexDev404** [comment]
@ItzLevvie 
https://github.com/microsoft/winget-pkgs/pull/61780#issuecomment-1138842988

**Review by ItzLevvie — APPROVED** [review]
> Does this mean I have to contact each one of them individually?

Yes.

**Comment by AlexDev404** [comment]
> > Does this mean I have to contact each one of them individually?
> @ItzLevvie 
> Yes.

Ok, I will do that. I have already contacted McAfee, AVG, Avast and Kaspersky but it seems as though Microsoft Defender is also detecting this as malware. 

Do I have to contact you all directly to remove the false-positive? Or can I just report it here?

**Comment by AlexDev404** [comment]
> Hello @AlexDev404, I am sorry to report that the Sha256 Hash does not match the installer. This maybe caused by using a vanity URL rather than a URL directly to the binary.
> 
> Please check the HASH and submit again. Sorry for the inconvenience.

New version of Metasploit was released a couple days ago, that is why.

**Comment by AlexDev404** [comment]
> > but it seems as though Microsoft Defender is also detecting this as malware.
> 
> You can use https://www.microsoft.com/en-us/wdsi/filesubmission for that.

Thanks Levvy! :)

**Comment by AlexDev404** [comment]
UPDATE: I have got both McAfee and Kaspersky to remove the false-positive. Only Microsoft, AVG and Avast are left.

**Comment by AlexDev404** [comment]
@ItzLevvie By any chance do you know the detection name of the package? I currently have McAfee installed and cant scan it with Windows Defender.

**Comment by AlexDev404** [comment]
@ItzLevvie 
Hi again Levvie,

It seems as though I'm experiencing difficulties with some anti-malware vendors (Kaspersky, especially)...
So what I'm going to do is kindly ask if possible, if you could leave the PR open while I sort things out.

Once I've finished arranging things, **I will ping you.**

Thanks once again for your patience.

**Comment by AlexDev404** [comment]
@ItzLevvie Microsoft still hasnt gotten back to me BTW while all the rest of A/V vendors have. 

What next?

**Comment by ItzLevvie** [comment]
> What next?

You'll have to wait until it's cleared out - sometimes it can take several weeks or months.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
You can set this pull request to DRAFT.

**Comment by AlexDev404** [comment]
> > What next?
> 
> You'll have to wait until it's cleared out - sometimes it can take several weeks or months.

Ah, alright then. I guess that's what we'll have to do

**Comment by AlexDev404** [comment]
> You can set this pull request to DRAFT.

@ItzLevvie I'll be setting all the rest of my unmerged pulls to that from now on

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by ItzLevvie** [comment]
[Policy] Area-External

**Pull request closed without merging**